File: app/src/pfp/java/social/entourage/android/map/filter/MapFilterFactory.java
package social.entourage.android.map.filter;

import android.content.Context;

import social.entourage.android.EntourageApplication;
import social.entourage.android.authentication.AuthenticationController;

/**
 * Created by mihaiionescu on 27/10/16.
 */

public class MapFilterFactory {

    public static MapFilter getMapFilter(boolean isProUser) {
        return new MapFilter();
    }

    public static MapFilter getMapFilter() {
        EntourageApplication app = EntourageApplication.get();
        if (app != null && app.getEntourageComponent() != null) {
            AuthenticationController authenticationController = app.getEntourageComponent().getAuthenticationController();
            if (authenticationController != null) {
                return authenticationController.getMapFilter();
            }
        }
        return new MapFilter();
    }

}

File: app/src/pfp/java/social/entourage/android/map/filter/MapFilterFactory.kt
package social.entourage.android.map.filter

import social.entourage.android.EntourageApplication

/**
 * Created by mihaiionescu on 27/10/16.
 */
object MapFilterFactory {
    @JvmStatic
    fun getMapFilter(isProUser: Boolean): MapFilter {
        return MapFilter()
    }

    @JvmStatic
    val mapFilter: MapFilter
        get() {
            return EntourageApplication.get()?.entourageComponent?.authenticationController?.mapFilter ?: MapFilter()
        }
}